torch/storage.py

```diff
@@ -192,18 +192,23 @@ def complex_float(self):
         return self._to(torch.cfloat)
 
     def pin_memory(self, device="cuda"):
-        """Copies the storage to pinned memory, if it's not already pinned."""
-        if self.is_cuda:
+        """Copies the CPU storage to pinned memory, if it's not already pinned."""
+        if self.device.type != 'cpu':
             raise TypeError(f"cannot pin '{self.type()}' only CPU memory can be pinned")
-        """For other backends, device need to set.
-           If not set, the default behaviour is CUDA device. Currently, only CUDA is supported.
-        """
-        if device != "cuda":
-            raise TypeError(f"cannot pin memory to {device} device, only CUDA is supported.")
 
-        import torch.cuda
-        allocator = torch.cuda.memory._host_allocator()  # type: ignore[attr-defined]
-        return type(self)(self.size(), allocator=allocator).copy_(self)
+        """For other backends, device need to set. If not set, the default behaviour is CUDA device."""
+        if device == "cuda":
+            import torch.cuda
+            allocator = torch.cuda.memory._host_allocator()  # type: ignore[attr-defined]
+            return type(self)(self.size(), allocator=allocator).copy_(self)
+        else :
+            import torch
+            if torch._C._get_privateuse1_backend_name() == device:
+                pinned_tensor = torch.tensor([], dtype=torch.uint8, device=self.device).set_(
+                    cast(Storage, self)).pin_memory(device)
+                return pinned_tensor._typed_storage()._untyped_storage
```

Comment:
you can access the untyped storage directly with `pinned_tensor.untyped_storage()`

Reply:
Yeah, it's really a better way, I get it now.